pkg/destroy/nutanix/nutanix.go

```diff
@@ -0,0 +1,229 @@
+	key    string
+	values []string
+}
+
+// New returns an Nutanix destroyer from ClusterMetadata.
+func New(logger logrus.FieldLogger, metadata *installertypes.ClusterMetadata) (providers.Destroyer, error) {
+	v3Client, err := nutanixtypes.CreateNutanixClient(context.TODO(),
+		metadata.ClusterPlatformMetadata.Nutanix.PrismCentral,
+		metadata.ClusterPlatformMetadata.Nutanix.Port,
+		metadata.ClusterPlatformMetadata.Nutanix.Username,
+		metadata.ClusterPlatformMetadata.Nutanix.Password,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clusterUninstaller{
+		clusterID: metadata.ClusterID,
+		infraID:   metadata.InfraID,
+		v3Client:  v3Client,
+		logger:    logger,
+	}, nil
+}
+
+// Run is the entrypoint to start the uninstall process.
+func (o *clusterUninstaller) Run() (*installertypes.ClusterQuota, error) {
+	o.logger.Infof("Starting deletion of Nutanix infrastructure for Openshift cluster %s", o.infraID)
+	err := wait.PollImmediateInfinite(time.Second*30, o.destroyCluster)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to destroy cluster")
+	}
+
+	return nil, nil
+}
+
+func (o *clusterUninstaller) destroyCluster() (bool, error) {
+	cleanupFuncs := []struct {
+		name    string
+		execute func(*clusterUninstaller) error
+	}{
+		{name: "VMs", execute: cleanupVMs},
+		{name: "Images", execute: cleanupImages},
+		{name: "Categories", execute: cleanupCategories},
+	}
+
+	done := true
+	for _, cleanupFunc := range cleanupFuncs {
+		if err := cleanupFunc.execute(o); err != nil {
+			o.logger.Debugf("%s: %q", cleanupFunc.name, err)
+			done = false
+		}
+	}
+	return done, nil
+}
+
+func cleanupVMs(o *clusterUninstaller) error {
+	ec := createExpectedCategory(o.infraID)
+	matchedVirtualMachineList := make([]*nutanixclientv3.VMIntentResource, 0)
+	allVMs, err := o.v3Client.V3.ListAllVM(emptyFilter)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range allVMs.Entities {
+		if hasCategoryAssigned(v.Metadata, ec) {
+			matchedVirtualMachineList = append(matchedVirtualMachineList, v)
+		}
+	}
+
+	if len(matchedVirtualMachineList) == 0 {
+		o.logger.Infof("no VMs found that require deletion for cluster %s", o.clusterID)
+	} else {
+		logToBeDeletedVMs(matchedVirtualMachineList, o.logger)
+		err := deleteVMs(o.v3Client.V3, matchedVirtualMachineList, o.logger)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func cleanupImages(o *clusterUninstaller) error {
+	ec := createExpectedCategory(o.infraID)
+	allImages, err := o.v3Client.V3.ListAllImage(emptyFilter)
+	if err != nil {
+		return err
+	}
+
+	allErrs := field.ErrorList{}
+	for _, image := range allImages.Entities {
+		if hasCategoryAssigned(image.Metadata, ec) {
+			imageName := *image.Spec.Name
+			imageUUID := *image.Metadata.UUID
+			o.logger.Infof("deleting image %s with UUID %s", imageName, imageUUID)
+			response, err := o.v3Client.V3.DeleteImage(imageUUID)
+			if err != nil {
+				o.logger.Errorf("failed to delete image %s: %q", imageUUID, err)
```

Comment:
Use `%v` for formatting the error. You don't want to wrap the entire error in quotes. The `%q` would be good for the image UUID, however.
```suggestion
				o.logger.Errorf("failed to delete image %q: %v", imageUUID, err)
```